src/databases/any.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{databases::DatabasePool, SessionError};
+use async_trait::async_trait;
+use std::fmt::Debug;
+use std::sync::Arc;
+
+/// [SessionAnyPool] is effectively a `dyn DatabasePool`. It can be useful if your application
+/// requires a runtime decision between multiple database backends. For example using `sqlite`
+/// in development builds but `postgres` in production builds.
+#[derive(Clone)]
+pub struct SessionAnyPool {
+    pool: Arc<dyn DatabasePool + Send + Sync>,
+}
+
+impl SessionAnyPool {
+    pub fn new(pool: Arc<dyn DatabasePool + Send + Sync>) -> Self {
```

Comment:
this should be able to get changed directly to the data type without the Arc within the parameters.
then inside of the new function it should Arc::new(pool) for them. if it doesn't work without the Arc then try making it a Generic that requires the traits instead.

Reply:
Couldn't get it to work without the Arc, the compiler complains about the `new` function which can't take an unsized `dyn` parameter. I also tried making it `impl DatabasePool`, which should be similar to the generic solution that you proposed, but that didn't work either. Or were you thinking about a different way?